wiki/installing-and-updating-drivers.md

```diff
@@ -0,0 +1,206 @@
+## Troubleshooting 🔧
+### If you can access Windows:
+1. Open the Device Manager by pressing <kbd>Win</kbd> + <kbd>X</kbd> and then clicking the "Device Manager" option. You can also search for it in the Start Menu.
+
+<img src="./img/installing-and-updating-drivers/troubleshooting/device-manager-power-user-menu.png" alt="The Win+X 'power user' menu, with Device Manager highlighted." width="200px">
+
+2. In Device Manager, find the device that is causing problems (you might have to expand a category), right-click on it, and click "Properties".
+
+<img src="./img/installing-and-updating-drivers/troubleshooting/device-manager-properties-button.png" alt="The context menu for a display adapter in Device Manager, with Properties highlighted." width="500px">
+
+3. Go to the "Driver" tab at the top of the window, and then click "Roll Back Driver".
+
+<img src="./img/installing-and-updating-drivers/troubleshooting/device-manager-properties-roll-back-driver.png" alt="The properties window for a device in Device Manager, with the Driver tab and Roll Back Driver button highlighted." width="500px">
+
+4. Windows will ask you why you're rolling back to a previous driver. Select a reason, and click "Yes". If you want to, you can leave a detailed response in the "Tell Us More" field, at the bottom of the window.
+
+<img src="./img/installing-and-updating-drivers/troubleshooting/driver-rollback-confirmation.png" alt="The confirmation dialogue presented after selecting Roll Back Driver." width="500px">
+
+5. Windows will then restore your driver to the previous version, which could take up to 5-10 minutes.
+
+<img src="./img/installing-and-updating-drivers/troubleshooting/hardware-change-restart-prompt.png" alt="The prompt to restart your computer, shown after rolling back a driver." width="500px">
+
+### Safe Mode
+Safe mode is a diagnostic tool, which loads Windows with only the essential drivers. This allows you to access Windows even if the problematic driver is causing issues during normal boot.
+1. Enter Safe Mode:
+
+
+On Windows 10 or 11:
+
+- Open the Start menu, click on the power button, and hold <kbd>Shift</kbd> while clicking Restart. This will take you to the Windows Recovery Environment:
+
+<img src="./img/installing-and-updating-drivers/winre/winre.png" alt="A screenshot of the Windows Recovery Environment." height="350px">
+
+Then go throught the steps:
+ - Troubleshoot
+<img src="./img/installing-and-updating-drivers/winre/troubleshoot.png" alt="An image of advanced startup" height="350px">
+ - Advanced options
+<img src="./img/installing-and-updating-drivers/winre/advanced_options.png" alt="An image to entrace to advanced options" height="350px">
+ - Startup Settings > Restart. After your computer restarts, press <kbd>4</kbd> to boot into Safe Mode, or <kbd>5</kbd> to boot into Safe Mode with internet access.
+ <img src="./img/installing-and-updating-drivers/winre/startup_settings.png" alt="An image to click to startup options" height="350px">
```

Comment:
```suggestion
On Windows 10 or 11:

1. Open the Start menu, click on the power button, and hold <kbd>Shift</kbd> while clicking Restart. This will take you to the Windows Recovery Environment:


2. Click on "Troubleshoot"

   <img src="./img/installing-and-updating-drivers/winre/troubleshoot.png" alt="An image of advanced startup" height="350px">

3. Then, select "Advanced options"

   <img src="./img/installing-and-updating-drivers/winre/advanced_options.png" alt="An image to entrace to advanced options" height="350px">

4. Finally, click "Startup Settings" and then "Restart".
 <img src="./img/installing-and-updating-drivers/winre/startup_settings.png" alt="An image to click to startup options" height="350px">

   After your computer restarts, press <kbd>4</kbd> to boot into Safe Mode, or <kbd>5</kbd> to boot into Safe Mode with internet access.
```